docs/developing/deploy_facilities/interacting_with_spl_tokens.md

```diff
@@ -4,18 +4,90 @@ proofedDate: na
+
+| Contract            | Usage                                | Account requirements                               | Mint tx signed by                                                    |
+| ------------------- | ----------------------------------- | ------------------------------------------ | -------------------------------------------------------------------- |
+| ERC20ForSpl         | Wrap Solana-minted token | 1\. Neon Account<br></br>2\. Existing SPL token | Signed by the Solana account private key (via linked Phantom wallet) |
+| ERC20ForSplMintable | Mint a token on Solana   | 1\. Neon Account                           | Signed by the Neon account private key (via linked MetaMask wallet)  |
+
+
+
+<Tabs>
+ <TabItem value="Constructor non-mintable" label="ERC20ForSpl Constructor" default>
+
+```
+constructor(
+     bytes32 _tokenMint
+)
+Arguments:
+_tokenMint – address of SPL token account
+Constructor signature for Mintable token is:
+constructor(
+<!--      string memory _name,
+     string memory _symbol, Is this to be removed for non-mintable?? -->
+     uint8 _decimals,
+     address _mint_authority
+)
+Arguments:
+_name – string representing full name of the token 
+_symbol – string representing shorten symbol of the token 
+_decimals – decimals of new token
+_mint_authority – address of mint/freeze authority Neon account
+```
+ </TabItem>
+<TabItem value="Constructor mintable" label="ERC20ForSplMintable Constructor">
+
+Note that before setting up this contract, you must register the token's [Metaplex metadata](https://docs.metaplex.com/programs/token-metadata/overview).
+
+``` 
+constructor(
+     string memory _name,
+     string memory _symbol,
+     bytes32 _tokenMint
+)
+Arguments:
+_name – string representing full name of the token 
+_symbol – string representing shorten symbol of the token 
+_tokenMint – address of SPL token account
+Constructor signature for mintable token is:
+constructor(
+     string memory _name,
+     string memory _symbol,
+     uint8 _decimals,
+     address _mint_authority
+)
+Arguments:
+_name – string representing full name of the token 
+_symbol – string representing shorten symbol of the token 
+_decimals – decimals of new token
+_mint_authority – address of mint/freeze authority Neon account 
+```
+ </TabItem>
+</Tabs>
 
-Source code:
-  * [Rust source code](https://github.com/neonlabsorg/neon-evm/blob/c43345d7abf7af14aa840e6b15c0fc64b084bb2c/evm_loader/program/src/precompile_contracts.rs#L106)
-  * [Solidity wrapper source code](https://github.com/neonlabsorg/neon-evm/blob/6ac1734658f0fdcac09092bd98979d4f6fe4530d/evm_loader/solidity/erc20_for_spl.sol#LL12C11-L12C11)
 
 ### Contract Interface
```

Comment:
This needs to be rewritten.

Standard functions could be removed, we only need to document our extension over the normal ERC-20 interface.

_1. function approveSolana(bytes32 spender, uint64 amount) public returns (bool)_
_3. function transferSolana(bytes32 to, uint64 amount) public returns (bool)_
_4. function claim(bytes32 from, uint64 amount) external returns (bool)_
_5. function claimTo(bytes32 from, address to, uint64 amount) public returns (bool)_

Reply:
Confused -- lines 16--18 have been deleted
Are you asking for them to be returned and updated?